packages/block-serialization-default-parser/parser.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+function bsdp_parse($document ) {
+    static $parser;
+
+    if ( ! isset( $parser ) ) {
+        $parser = new BSDP_Parser();
```

Comment:
Agreed. `Block_Parser()`?

Reply:
mainly this is there to prevent namespace collisions. my hope is that a few PRs after this we'll have a filter choose the parser and obviously if we create two or more `Block_Parser()` classes we'll run into conflicts.

any thoughts on that? even with an encapsulating class we run into some issues here because I don't think we can create a class within a class. the only way around it otherwise I think is actual namespacing which isn't supported on older PHP version…

Reply:
Realistically, is there going to be a completely new parser appear between now and 5.0? It seems like this parser is going to be the one that will go into Core.

If that's the case, we should just use a generic name. `WP_Block_Parser` will fit into the WordPress naming scheme.